### ISSUE TITLE
run e2e tests and unit tests on Windows and macOS

### ISSUE THREAD

**Issue by kadel** [redhat-developer/odo#401]
Currently, we run our tests only on Linux we should run it also on Windows and macOS.
we can run macOS test on travis, and for windows we can use https://www.appveyor.com/ or our internal resources.

Windows is a priority right now.

**Comment by kadel** [comment]
Hi @ashetty1. We can put this on hold for now. Can you please just describe what problems you encountered co we have a record of it when we get beck to that.

**Comment by ashetty1** [comment]
Travis doesn't support docker with MacOS. Also, running docker would involve setting up virtualbox which is not really feasible on travis. About windoes, haven't been able to get the tests running.

**Comment by cdrage** [comment]
@ashetty1 I've used SemaphoreCI to bring up VirtualBox VM's for OpenShift and Kubernetes.

Unfortunately, it's against licensing to emulate / bring up Windows boxes as well as OS X :( We could (in the future) ask help from the CentOS CI team as IIRC they have some Windows boxes available for running Jenkins tests on.

**Comment by kadel** [comment]
It is time to start working on it. Especially the Windows

**Comment by ashetty1** [comment]
Tried `git config --system core.longpaths true`, but looks like admin access is not available for free users on appveyor.

**Comment by cdrage** [comment]
Update, https://github.com/redhat-developer/odo/pull/783 has been merged so now it's:

- [ ] Running the actual tests (since it has now been setup)

**Comment by kadel** [comment]
travis-ci added windows guests https://blog.travis-ci.com/2018-10-11-windows-early-release
we should investigate and try to use to to run e2e on windows

**Comment by kadel** [comment]
It would be great to create one uniform e2e test that can run on all three platforms

**Comment by girishramnani** [comment]
assuming we are planning to run windows VMs on Centos CI, this gets blocked by the centos CI onboarding issue @syamgk

**Comment by amitkrout** [comment]
I guess this is an umbrella issue comprises 

1. #1131
2. #1129

Currently our test code is designed to be run on only linux like platform. So the issue 1 and 2 makes all the integration test code platform independent. I am close to fix the Issue #1131. I would be happy if @syamgk shares some of his capacity to fix #1129. 

I think getting the platform is the key issue as part of unit tests on macOS is concerned. @syamgk correct me if i am wrong.

**Comment by syamgk** [comment]
> assuming we are planning to run windows VMs on Centos CI, this gets blocked by the centos CI onboarding issue @syamgk

no that assumption is wrong, we have option to enable windows machine test on travis-ci and the plan is to stick with it until we get into central ci or any other place with windows setup

**Comment by syamgk** [comment]
> I guess this is an umbrella issue comprises
> 
>     1. #1131
> 
>     2. #1129
> 
> 
> Currently our test code is designed to be run on only linux like platform. So the issue 1 and 2 makes all the integration test code platform independent. I am close to fix the Issue #1131. I would be happy if @syamgk shares some of his capacity to fix #1129.
> 
> I think getting the platform is the key issue as part of unit tests on macOS is concerned. @syamgk correct me if i am wrong.

macOS is available on travis-ci we had a test which ran on mac for building odo we removed it in november i guess. https://docs.travis-ci.com/user/reference/osx

**Comment by cdrage** [comment]
I'm going to make this medium priority for now.

**Comment by cdrage** [comment]
Have to unassign myself to this for now.

Ping @amitkrout / @mohammedzee1000

**Comment by prietyc123** [comment]
- [ ] #2540 
- [ ] #2539

**Comment by kadel** [comment]
/close 
in favor of https://github.com/openshift/odo/issues/2540
